### PR TITLE
Status cmd is working - still have a few TODOs

### PR DESCRIPTION
**Pull request by jpricketMSFT**
User Story #884931

### CODE REVIEW

File: src/tfvc/commands/commandhelper.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+"use strict";
+
+import { parseString } from "xml2js";
+
+export class CommandHelper {
+    public static async parseXml(xml: string): Promise<any> {
```
Comment (jeffyoung) [inline src/tfvc/commands/commandhelper.ts]: CAP-it

File: src/tfvc/commands/commandhelper.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    private static normalizeName(name: string): string {
+        if (name) {
+            return name.replace(/\-/g, "").toLowerCase();
+        }
+        return name;
+    }
+
+    public static trimToXml(xml: string): string {
```
Comment (jeffyoung) [inline src/tfvc/commands/commandhelper.ts]: CAP-it

File: src/tfvc/commands/commandhelper.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+"use strict";
+
+import { parseString } from "xml2js";
+
+export class CommandHelper {
+    public static async parseXml(xml: string): Promise<any> {
+        return new Promise<any>((fulfill, reject) => {
+            parseString(xml, {
```
Comment (jeffyoung) [inline src/tfvc/commands/commandhelper.ts]: I'd push the {  to the next line and align with the 'x' in xml, then...

File: src/tfvc/commands/commandhelper.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                            tagNameProcessors: [CommandHelper.normalizeName],
+                            attrNameProcessors: [CommandHelper.normalizeName]
+                        },
+                        (err, result) => {
+                if (err) {
```
Comment (jeffyoung) [inline src/tfvc/commands/commandhelper.ts]: Can you indent the "if (err) {" or is tslint down with it like this?

File: src/tfvc/commands/commandhelper.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                    reject(err);
+                } else {
+                    fulfill(result);
```
Comment (jeffyoung) [inline src/tfvc/commands/commandhelper.ts]: I'm not sure of the diff between fulfill and resolve.  I use "resolve".

File: src/tfvc/uihelper.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                items.push({
+                    label: UIHelper.GetFileName(changes[i]),
+                    description: changes[i].changeType,
+                    detail: UIHelper.GetRelativePath(changes[i])
+                    });
+            }
+            // Then, show the quick pick window and get back the one they chose
+            //TODO localize strings
+            let item: QuickPickItem = await window.showQuickPick(
+                items, { matchOnDescription: true, placeHolder: "Choose a file to open the file." });
```
Comment (jeffyoung) [inline src/tfvc/uihelper.ts]: At some point, we'll have to put these hard-coded strings into a Strings file.

File: src/tfvc/uihelper.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+            // Finally, find the matching pending change and return it
+            if (item) {
+                for (let i = 0; i < changes.length; i++) {
+                    if (UIHelper.GetRelativePath(changes[i]) === item.detail) {
+                        return changes[i];
+                    }
+                }
+            }
+        }
+        return null;
```
Comment (jeffyoung) [inline src/tfvc/uihelper.ts]: How's about either undefined or void 0?

File: src/tfvc/commands/status.ts [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+     * </status>
+     */
+    public async ParseOutput(executionResult: IExecutionResult): Promise<IPendingChange[]> {
+        let changes: IPendingChange[] = [];
+        const xml = CommandHelper.trimToXml(executionResult.stdout);
+        // Parse the xml using xml2js
+        const json = await CommandHelper.parseXml(xml);
+        if (json && json.status) {
+            // get all the pending changes first
+            const pending = json.status.pendingchanges[0].pendingchange;
```
Comment (jeffyoung) [inline src/tfvc/commands/status.ts]: Is there a data type for pending?

Comment (jeffyoung) [inline src/tfvc/commands/status.ts]: Do that, if you can, for all of your other variable definitions.

Comment (jpricketMSFT) [inline src/tfvc/commands/status.ts]: Had to use any :(

File: src/tfvc/uihelper.ts (hunk 6 above)
Comment (jpricketMSFT) [inline src/tfvc/uihelper.ts]: Agreed. Just trying to let the design settle a bit before doing that. Let's talk about how it should be done on Monday.

**Pull request merged**